src/common/extensions/map.ts

```diff
@@ -0,0 +1,29 @@
+import Guard from "../guard";
+
+/**
+ * Processes items in the map within the specified batch size (default: 5)
+ * @param this The map to process
+ * @param action The action to perform on each item in the map
+ * @param batchSize The batch size for actions to perform in parallel (default: 5)
+ */
+export async function forEachAsync<K, V>(
+    this: Map<K, V>,
+    action: (value: V, key: K) => Promise<void>,
+    batchSize: number = 5): Promise<void> {
+    Guard.null(this);
+    Guard.null(action);
+    Guard.expression(batchSize, (value) => value > 0);
+
+    const all: Array<[K, V]> = [...this.entries()];
```

Comment:
Again, same comment here

Reply:
Same - We don't want to mutate the original map

Reply:
Replied above